### PR TITLE
Fix target resolution when pointing to shadow DOM element.

### PR DESCRIPTION
**Pull request by gghez**
Hi guys, I don't know if that's something planned, but this is a fix I need in my current project using Polymer elements.

Thanks.

Issue: #159

### CODE REVIEW

**Comment by gghez** [comment]
CLA completed.

File: paper-char-counter.html [hunk 1]
```diff
@@ -94,13 +94,27 @@
...
+    _initTarget: function (targetElement) {
+        this._targetElement = targetElement;
+
+        this._maxChars = this._targetElement.maxLength;
+        this._targetElement.addEventListener('input', this.inputAction.bind(this));
+    },
+
+    targetChanged: function (o, n) {
+        if (n && !this._targetElement) {
+            this._initTarget(typeof n == 'string' ? document.getElementById(n) : n);
```
Comment (gghez) [inline paper-char-counter.html]: this allow writing:
```html
<paper-char-counter target="{{ $.myInput }}"></paper-char-counter>
```

**Comment by gghez** [comment]
@morethanreal does this look ok to merge?

File: paper-char-counter.html (hunk 1 above)
Comment (zerodevx) [inline paper-char-counter.html]: maybe you could add :
```javascript
this._charCount = this._targetElement.value.length;
```
between line 109 and 110? This populates the char counter if the target input element already has a predefined value.

Comment (gghez) [inline paper-char-counter.html]: You are right, but I throw in the towel regarding to small amount of interactions between project leads and external PR... sorry. By the way I switched back to AngularJS v1, too much fork were required to make polymer used in production.

**Comment by alechewitt** [comment]
any idea when this will be merged?
I'm using the char counter inside an angular directive and can only get it working using the code in your pull request.

**Comment by alechewitt** [comment]
I had a go at using your code with the paper-char-counter when in an angular directive. However although working I found it was still producing errors in the console.

I've amended it to look like this https://github.com/alechewitt/paper-input/blob/master/paper-char-counter.html#L107 which seems to prevent the errors.

**Pull request closed without merging**